lib/python-sdk/common_grants_sdk/client/client.py

```diff
@@ -0,0 +1,128 @@
+"""Main HTTP client for the CommonGrants API."""
+
+import httpx
+from typing import Optional
+from uuid import UUID
+
+from .auth import Auth
+from .config import Config
+from .opportunities import OpportunitiesListIterator
+from ..schemas.pydantic.models import OpportunityBase
+from ..schemas.pydantic.responses import OpportunityResponse, OpportunitiesListResponse
+
+
+class Client:
+    """HTTP client for interacting with the CommonGrants API."""
+
+    def __init__(
+        self,
+        config: Optional[Config] = None,
+        auth: Optional[Auth] = None,
+    ):
+        """Initialize the CommonGrants client.
+
+        Args:
+            config: Optional Config instance
+            auth: Optional Auth instance
+        """
+        self._config = config or Config()
+        self._auth = auth or Auth.api_key(self._config.api_key)
+        self._http_client = httpx.Client(timeout=self._config.timeout)
+
+    def _url(self, path: str) -> str:
+        """Construct a full URL from base URL and path.
+
+        Args:
+            path: URL path (should start with /)
+
+        Returns:
+            Complete URL string
+        """
+        base = self._config.base_url.rstrip("/")
+        return f"{base}{path}"
+
+    def list_opportunities(
+        self,
+        paginate: bool = False,
+        total: Optional[int] = None,
+        page: Optional[int] = None,
+        page_size: Optional[int] = None,
```

Comment:
I'm wondering if we want a default `max_items` or `max_requests` or something similar? 

If a system like Simpler.Grants.gov has thousands of records and we're fetching all of them in increments of 100, this could quickly run up someone's usage limits without realizing it.

I think SGG has fairly high limits, but giving consumers some control over their usage could be helpful. Maybe we can look at other APIs that have built-in pagination and see if they offer a similar way to break out of several round trips.